### PR TITLE
[CB-82] Schema Strategy

### PR DESCRIPTION
**Pull request by EPecherkin**

### CODE REVIEW

File: lib/crystalball/rails/predictor/modified_schema.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          if revision
+            repository.lib.show(revision, SCHEMA_PATH)
+          else
+            File.read(File.join(repository.dir.path, SCHEMA_PATH))
+          end
+        end
+
+        def changed_tables(schema1, schema2)
+          changed_tables = []
+          schema1.each do |table_name, body|
```
Comment (jaimerson) [inline lib/crystalball/rails/predictor/modified_schema.rb]: `schema1.map`?

File: spec/rails/helpers/schema_definition_parser_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    before do
+      allow(described_class).to receive(:new).and_return(parser)
+    end
+
+    it 'returns hash with schema info' do
+      expect(subject).to eq(
+        'users' => {
+          options: [
+            [{force: :cascade}],
+            ['users', {column: 'name'}]
```
Comment (pyromaniac) [inline spec/rails/helpers/schema_definition_parser_spec.rb]: Not really clear if this is an FK

**Review by pyromaniac — APPROVED** [review]

File: lib/crystalball/rails/tables_map_generator.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+      # @return [Crystalball::Rails::TablesMap]
+      def map
+        @map ||= TablesMap.new(metadata: {commit: configuration.commit, version: configuration.version})
+      end
+
+      private
+
+      def repo
+        @repo = GitRepo.open('.') unless defined?(@repo)
```
Comment (jaimerson) [inline lib/crystalball/rails/tables_map_generator.rb:64]: Why not `||=`?

File: spec/rails/helpers/schema_definition_parser_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      end
+       add_foreign_key 'users', 'users', column: 'name'
+    end
+  RUBY
+
+  describe '.parse' do
+    subject { described_class.parse(schema) }
+
+    before do
+      allow(described_class).to receive(:new).and_return(parser)
```
Comment (jaimerson) [inline spec/rails/helpers/schema_definition_parser_spec.rb:21]: Is that necessary?

**Review by jaimerson — APPROVED** [review]
:+1:

File: lib/crystalball/rails/tables_map_generator.rb (hunk 3 above)
Comment (EPecherkin) [reply lib/crystalball/rails/tables_map_generator.rb:64]: I'm not sure. I took it from MapGenerator
@pluff you should know I think

File: spec/rails/helpers/schema_definition_parser_spec.rb (hunk 4 above)
Comment (EPecherkin) [reply spec/rails/helpers/schema_definition_parser_spec.rb:21]: Yes

**Pull request merged**